src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1057,17 +1060,53 @@ func (c *controller) updateNodesLabels() bool {
 			continue
 		}
 		hostsWithLabels[hostname] = hostData
+		// if host is worker and new role was set we need to pause MCP
+		if hostData.Host.Role == models.HostRoleWorker && strings.Contains(hostData.Host.NodeLabels, roleLabel) {
```

Comment:
MCP has field match labels.  It contains not just roles.
We should find out if any of the requested labels by the MCP are defined on the node.

Reply:
We can do it another way, pause worker pool if host has mcp pool and not role label
WDYT?